.github/workflows/build-webapi.yaml

```diff
@@ -55,15 +55,46 @@ jobs:
         with:
           directory: ${{ env.DOTNET_SRC_DIR }}
       
-      - name: Generate Open API
-        run: dotnet tool run swagger tofile --output ./api/v1/swagger.yaml --yaml ${{ env.DLL_FILE_PATH }} v1  
+      - name: Generate Branch Open API
+        run: dotnet tool run swagger tofile --output ./api/v1/pr-swagger.yaml --yaml ${{ env.DLL_FILE_PATH }} v1  
         working-directory: ./src/CarbonAware.WebApi/src 
       
-      - name: Upload artifact
+      - name: Upload branch artifact
         uses: actions/upload-artifact@v1
         with:
           name: pr-swagger.yaml
-          path: src/CarbonAware.WebApi/src/api/v1/swagger.yaml
+          path: src/CarbonAware.WebApi/src/api/v1/pr-swagger.yaml
+
+      - name: Checkout Dev Branch
```

Comment:
Shouldn't be these steps a different job?

Reply:
I agree with @juzuluag -- the better place to do this, is in the "api-comparison" job, which already checks-out the dev branch, and already builds a "dev-branch-based-swagger" file from it -- and then downloads the "PR-branch-based-swagger" file from the prior job -- I would recommend uploading the additional `dev-swagger.yaml` build-artifact after the "Generate Open API" step in the "api-comparison" job